### PR TITLE
Package the java-matter-controller as an executable format

### PR DESCRIPTION
**Pull request by yufengwangca**
Currently, we need to run the java-matter-controller with the following format 

`java -verbose:jni -Xcheck:jni -Djava.library.path=out/linux-x64-java-matter-controller/lib/jni -cp out/linux-x64-java-matter-controller/lib/third_party/connectedhomeip/src/controller/java/CHIPController.jar:out/linux-x64-java-matter-controller/bin/java-matter-controller com.matter.controller.Main `

We need to package it in an executable format to be run like
 $ java -Djava.library.path=../lib/jni -jar java-matter-controller

### CODE REVIEW

**Review by yunhanw-google — APPROVED** [review]

File: examples/java-matter-controller/Manifest.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 Main-Class: com.matter.controller.Main
+Class-Path: ../lib/third_party/connectedhomeip/src/controller/java/CHIPController.jar
```
Comment (yunhanw-google) [inline examples/java-matter-controller/Manifest.txt:2]: SetupPayloadParser

Comment (yufengwangca) [reply examples/java-matter-controller/Manifest.txt]: done

File: scripts/build/builders/host.py [hunk 2]
```diff
@@ -483,6 +432,9 @@ def PostBuildCommand(self):
             self._Execute(['genhtml', os.path.join(self.coverage_dir, 'lcov_final.info'), '--output-directory',
                            os.path.join(self.coverage_dir, 'html')], title="HTML coverage")
 
+        if self.app.ExamplePath() == "java-matter-controller" and 'JAVA_PATH' in os.environ:
```
Comment (andy31415) [inline scripts/build/builders/host.py:435]: why the environment test?

for the test use `self.app == HostApp.JAVA_MATTER_CONTROLLER`

File: examples/java-matter-controller/java/src/com/matter/controller/Main.java [hunk 3]
```diff
@@ -82,14 +85,27 @@ private static void registerCommandsPairing(
   }
 
   public static void main(String[] args) {
-    /* TODO: uncomment when SDK integration is done
+    System.setProperty("java.library.path", "../lib/jni");
```
Comment (andy31415) [inline examples/java-matter-controller/java/src/com/matter/controller/Main.java:88]: is this temporary? Should this be gated by some main argument flag?

File: examples/java-matter-controller/java/src/com/matter/controller/Main.java [hunk 4]
```diff
@@ -82,14 +85,27 @@ private static void registerCommandsPairing(
...
+
+    Field fieldSysPath = null;
+    try {
```
Comment (andy31415) [inline examples/java-matter-controller/java/src/com/matter/controller/Main.java:91]: Some comments on what this does and why would help.

I could find some description of this at http://stackoverflow.com/a/24988095  ... but do we want this? At least we could comment for alternatives possible - I see many java apps have some form of launcher that set the right paths.

Comment (yufengwangca) [reply examples/java-matter-controller/java/src/com/matter/controller/Main.java]: From the stackoverflow thread, it looks setting the java.library.path directly within the Java program is a workaround

Remove this part and use java -Djava.library.path=../lib/jni -jar java-matter-controller to set the correct natives directory at this moment.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**